Index: deepin-libvirt/src/remote/libvirtd.policy
===================================================================
--- deepin-libvirt.orig/src/remote/libvirtd.policy
+++ deepin-libvirt/src/remote/libvirtd.policy
@@ -28,6 +28,11 @@ License along with this library.  If not
     <action id="org.libvirt.unix.monitor">
       <description>Monitor local virtualized systems</description>
       <message>System policy prevents monitoring of local virtualized systems</message>
+      <message xml:lang="zh_CN">打开虚拟机需要管理员权限</message>
+      <message xml:lang="zh_TW">啟動虛擬機需要管理員權限</message>
+      <message xml:lang="zh_HK">啟動虛擬機需要管理員權限</message>
+      <message xml:lang="ja_JP">仮想マシンを起動するには管理者権限が必要です</message>
+      <message xml:lang="fr_FR">Lancer une machine virtuelle nécessite des privilèges administrateur</message>
       <defaults>
         <!-- Any program can use libvirt in read-only mode for monitoring,
              even if not part of a session -->
@@ -40,6 +45,11 @@ License along with this library.  If not
     <action id="org.libvirt.unix.manage">
       <description>Manage local virtualized systems</description>
       <message>System policy prevents management of local virtualized systems</message>
+      <message xml:lang="zh_CN">打开虚拟机需要管理员权限</message>
+      <message xml:lang="zh_TW">啟動虛擬機需要管理員權限</message>
+      <message xml:lang="zh_HK">啟動虛擬機需要管理員權限</message>
+      <message xml:lang="ja_JP">仮想マシンを起動するには管理者権限が必要です</message>
+      <message xml:lang="fr_FR">Lancer une machine virtuelle nécessite des privilèges administrateur</message>
       <defaults>
         <!-- Any program can use libvirt in read/write mode if they
              provide the root password -->
